types/api.yaml

```diff
@@ -31,19 +31,31 @@ ValidatorBalanceResponse:
 ValidatorStatus:
   description: |
     Possible statuses:
-    - **pending_initialized** - When the first deposit is processed, but not enough funds are available (or not yet the end of the first epoch) to get validator into the activation queue.
-    - **pending_queued** - When validator is waiting to get activated, and have enough funds etc. while in the queue, validator activation epoch keeps changing until it gets to the front and make it through (finalization is a requirement here too).
-    - **active_ongoing** - When validator must be attesting, and have not initiated any exit.
-    - **active_exiting** - When validator is still active, but filed a voluntary request to exit.
-    - **active_slashed** - When validator is still active, but have a slashed status and is scheduled to exit.
-    - **exited_unslashed** - When validator has reached reguler exit epoch, not being slashed, and doesn't have to attest any more, but cannot withdraw yet.
-    - **exited_slashed** - When validator has reached reguler exit epoch, but was slashed, have to wait for a longer withdrawal period.
-    - **withdrawal_possible** - After validator has exited, a while later is permitted to move funds, and is truly out of the system.
-    - **withdrawal_done** - (not possible in phase0, except slashing full balance) - actually having moved funds away
+    - **waiting_for_eligibility** - A `Deposit` is included in a `BeaconBlock`, causing the Beacon Chain to become aware of the validator.
+    - **waiting_for_finality** - The Beacon Chain has observed that the validator could become eligible and it is waiting for those conditions 
+    to become finalized.
+    - **waiting_in_queue** - The conditions which allow the validator to become active are finalized.
+    The validator is now waiting in the activation queue, awaiting its turn to be activated.
+    - **standby_for_active** - The conditions which allow the validator to become active are finalized.
+    The validator is now waiting in the activation queue, awaiting its turn to be activated.
+    Validator status will change to `active` on `validator.activation_epoch`.
+    - **active** - The validator is now active and, for the first time, is rewarded for including attestations and blocks in the beacon chain.
+     It will be penalized if it fails to have attestations included in the chain.
+    - **active_awaiting_slashed_exit** - The validator has produced conflicting messages and they have been included
+     in the chain via a ProposerSlashing or AttesterSlashing. It will not be rewarded for producing attestations but may be 
+     required to (and rewarded for) producing blocks. Validator will change status to `exited_slashed` on `validator.exit_epoch`.
+    - **active_awaiting_voluntary_exit** - The validator has voluntarily started exiting. As with the active state, the
```

Comment:
@paulhauner One thing to note about your DAG is that you can *kind of* go from `active_awaiting_voluntary_exit` to `active_awaiting_slashed_exit`. Well I suppose you voluntarily exited but are now slashed..

Similarly, you can be `exited_voluntarily` and then slashed. Which what state does that put you in? `exited_slashed`? In such a case, the DAG should be updated to account for these scenarios

Reply:
A couple of note about "8. `exited_volutarily` & `exited_slashed`" in your doc. 

> This withdrawable epoch was decided by the protocol when the exit was initiated.

This is true in many cases, but if the validator first sent a voluntary exit and then was slashed, the withdrawable epoch would be updated to be longer once the val is slashed. So this epoch is actually decided "when the exit is initiated or the validator slashed, whichever was the latest of the two actions"

> The validator will remain in this state for 256 epochs (~27 hours) before
progressing to the next state.

This is the case when a non-slashed exit. Otherwise, the validator is not withdrawable until `slashing_epoch  + EPOCHS_PER_SLASHINGS_VECTOR` (36 days).

Reply:
I know those are a bit nit-picking but I want to better understand how you plan to represent these cases so we can try to conform on the edges